### PR TITLE
Onboarding - Add business extension recommendations

### LINKED ISSUE #2518: Onboarding - Profile Wizard: Add Facebook & MailChimp to the Business Tab
See p1561555686016700-slack-wc-onboarding and p1560865236005400-slack-wc-onboarding

The MailChimp and Facebook extensions should be added to the business tabs with toggles for the user to choose if they want them installed or not.

We should save this in the profiler data, because we'll need to pass these slugs along to WooCommerce.com to install.

### PR DESCRIPTION
**Pull request by justinshreve**
Fixes #2518.

This PR adds the Mailchimp and Facebook extensions to the business tab of the profile wizard.

Screen Shot 2019-07-01 at 4 54 59 PM

Note: I noticed certain values for `other_platform` are causing save endpoints to fail with a 400, even though they should be valid. This is not addressed in this PR. I can take a look at this tomorrow and fix in a follow-up PR.

### Detailed test instructions:

* Go to `/wp-admin/admin.php?page=wc-admin#/?step=business-details`
* Fill out the form and see that the extension recommendations display. Test the toggles, and saving your profile data. You can request `GET /wp-json/wc-admin/v1/onboarding/profile` to make sure the extensions saved.

### CODE REVIEW

File: client/dashboard/profile-wizard/steps/business-details.js [hunk 1]
```diff
@@ -83,6 +92,85 @@ class BusinessDetails extends Component {
...
+
+		const extensions = keys( pickBy( this.state.extensions ) );
+		if ( 0 === extensions.length ) {
+			return null;
+		}
+
+		return (
+			<p>
+				{ sprintf(
+					__( 'The following plugins will be installed for free: %s' ),
```
Comment (joshuatf) [inline client/dashboard/profile-wizard/steps/business-details.js]: 👍

File: client/dashboard/profile-wizard/style.scss [hunk 2]
```diff
@@ -213,6 +213,112 @@
...
+	}
+
+	.woocommerce-profile-wizard__product-types-card {
+		.woocommerce-profile-wizard__benefit-content {
+			margin-left: 12px;
+		}
+	}
+
+	.woocommerce-profile-wizard__tracking {
+		.components-form-toggle {
```
Comment (joshuatf) [inline client/dashboard/profile-wizard/style.scss]: My toggles look a little off-centered vertically:

Screen Shot 2019-07-02 at 10 20 41 AM

File: includes/api/class-wc-admin-rest-onboarding-profile-controller.php [hunk 3]
```diff
@@ -283,15 +283,27 @@ public static function get_profile_properties() {
 					'other',
 				),
 			),
-			'theme'           => array(
+			'business_extensions' => array(
```
Comment (joshuatf) [inline includes/api/class-wc-admin-rest-onboarding-profile-controller.php]: We'll need a couple quick changes in  `tests/api/onboarding-profile.php` to get tests ✅

Comment (justinshreve) [reply includes/api/class-wc-admin-rest-onboarding-profile-controller.php]: Done.

**Comment by justinshreve** [comment]
I've fixed the phpunit test.

@joshuatf I can't seem to replicate the off-center toggles. Your toggles look a bit bigger than mine, which could affect the positioning, so I did add a max height/width. Does that help? If not, could you see what CSS is causing that? I'm not having issues in Chrome or Safari.

**Comment by joshuatf** [comment]
@justinshreve Odd, I'm seeing it in all browsers.  I wonder if there's another plugin altering the behavior here.

>Your toggles look a bit bigger than mine

I think I may have been zoomed in.  Mine appear the same size as the ones in your screenshot.

The span here appears a bit off:

Screen Shot 2019-07-03 at 4 13 54 PM

But adding display `block` or `inline-block` to the toggle seems to position that span at the top:

`.components-form-toggle {
    display: inline-block;
}`

Screen Shot 2019-07-03 at 4 14 35 PM

**Comment by justinshreve** [comment]
@joshuatf Thanks -- I agree there must be some plugin difference (or maybe Gutenberg difference) because I already had that class coming in from wp components, but I've added that class in our CSS as well.

I've also adjusted when the extensions display so that the extensions don't disappear/reappear as you select from the selling dropdown (which happened after rebase now that we set a default onFocus).

**Comment by justinshreve** [comment]
> It feels like these styles should be included in @wordpress/components, but I guess that would create duplication between the WP included styles when updating. Is there some discussion around this already that I can read up on?

Good point... I agree. I did a quick search but don't see any discussion around that.

**Pull request merged**